packages/admin/admin/src/FinalFormSaveButton.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from "react";
+import { useForm, useFormState } from "react-final-form";
+import { FormattedMessage } from "react-intl";
+
+import { SaveButton } from "./common/buttons/save/SaveButton";
+import { messages } from "./messages";
+
+interface FinalFormSaveButtonProps {
+    message?: React.ReactNode;
+}
+
+export const FinalFormSaveButton = ({ message = <FormattedMessage {...messages.save} /> }: FinalFormSaveButtonProps) => {
+    const form = useForm();
+    const { pristine, hasValidationErrors, submitting, hasSubmitErrors } = useFormState();
+
+    return (
+        <SaveButton
+            disabled={pristine || hasValidationErrors || submitting}
+            color="primary"
+            variant="contained"
+            saving={submitting}
+            hasErrors={hasSubmitErrors}
+            onClick={() => {
+                !pristine && !hasValidationErrors && !submitting && form.submit();
+            }}
```

Comment:
```suggestion
            onClick={() => {
                form.submit();
            }}
```

The butt is disable when one of states is truthy, so we shouldn't need to check it here. We also don't do it in [FinalFormSaveSplitButton](https://github.com/vivid-planet/comet/blob/main/packages/admin/admin/src/FinalFormSaveSplitButton.tsx#L28)

Reply:
I disagree. Imo you shouldn't rely on element attributes to prevent something. It should always be prevented programmatically.

Reply:
Jup exactly. How would said person do it manually if the listener is programmatically blocked? He could maybe modify the JS bundle code 😂 

I agree that it makes sense to do it globally but I also think that this kind of inconsistency isn't that bad and could be tolerated. It doesn't change looks and UX.

Reply:
By not using your button, but submitting the form in another way: https://github.com/final-form/react-final-form/blob/master/docs/faq.md#how-can-i-trigger-a-submit-from-outside-my-form 😉

Reply:
Damn didn't know that^^

Reply:
You may keep them, but please move it into a variable 🙂